Sideload: create file in temp dir

akrantz opened the pull request:
Fixes issue https://github.com/OfficeDev/office-toolbox/issues/7

To allow the command to be repeatable and not clutter the source directory with new files, the sideload command is changed to create the document in the temp directory with filename "{app} add-in {id}" where app is the application host such as "excel" and {id} is the add-in guid from the manifest.

The same file path is used each time sideload is run. 

I was going to only create it if it didn't exist, but if the version is changed, the new version needs to be in the document. Therefore, the file is generated each time, deleting any existing file first.

akrantz committed:
sideload: create file in temp dir

akrantz commented on src/util.ts:
```diff
@@ -436,15 +436,15 @@ function generateTemplateFile(application: string, type: string, id: string, ver
+        fs.removeSync(templatePath);
       }
 
+      fs.ensureDirSync(path.dirname(templatePath));
```
I added this check in case the templatePath uses a subdirectory in the path. This could be added so that the sideload document files are in one directory such as "%TEMP%\office-toolbox-sideload". I'd like input if that would be a good addition, because then you could more easily delete that directory if desired. We don't have a way to cleanup these temp files. I was thinking that "npm stop" could invoke an office-toolbox command which would clean up, e.g. office-toolbox sideload-cleanup.

Ceespino commented:
@MsCesarBerard is added to the review. #Closed

Ceespino commented on src/util.ts:
```diff
@@ -436,15 +436,15 @@ function generateTemplateFile(application: string, type: string, id: string, ver
+      const templatePath = path.join(os.tmpdir(), `${application} add-in ${id}${extension}`);
+  
+      if (fs.existsSync(templatePath)) { 
+        fs.removeSync(templatePath);
```
>fs.removeSync(templatePath); [](start = 8, length = 28)

So as we discussed, we should handle the case when the file is open or locked and we are not able to delete it.

akrantz committed:
sideload: handle when file is in use

If the file is in use and cannot be deleted, try again with ".2", ".3", etc. appended to the filename (before the extension).

akrantz commented:
@MsCesarBerard I've updated this so that it handles when the file is in use. If it cannot delete the file, it adds ".2" to the filename (before the extension) and tries again. If still in use, then ".3", etc.

Ceespino reviewed (approved):
:shipit:
